src/BeadingStrategy/RedistributeBeadingStrategy.h

```diff
@@ -0,0 +1,61 @@
+     * 'right' number of walls but the wrong bead-widths (and locations) -- becasue it assumes the walls are some average. This is then
+     * rectified by the newly introduced meta-strategy, which redistributes the bead-widths. It takes a look at the factor of the thickness
+     * the outer and inner walls would have occupied in the ideal situation and applies that to the current situation. This keeps a) the
+     * original beadings simple to understand and debug b) the meanings and behaviour of the original beadings intact while still
+     * c) preserving the widths of the outer and inner walls the user (settings) has/have input at the very least in the case of 'full'
+     * walls (and does a decent attempt at at least preserving the relative relation between those two values with a lesser thickness).
+     */
+    class RedistributeBeadingStrategy : public BeadingStrategy
+    {
+    public:
+        /*!
+         * /param optimal_width_outer Outer wall width, guaranteed to be the actual (save rounding errors) at a bead count if the parent
+         *                            strategies' optimum bead width is a weighted average of the outer and inner walls at that bead count.
+         * /param optimal_width_outer Inner wall width, guaranteed to be the actual (save rounding errors) at a bead count if the parent
+         *                            strategies' optimum bead width is a weighted average of the outer and inner walls at that bead count.
+         */
+        RedistributeBeadingStrategy(coord_t optimal_width_outer, coord_t optimal_width_inner, BeadingStrategy* parent) :
+            BeadingStrategy(parent->optimal_width, parent->default_transition_length, parent->transitioning_angle),
+            parent(parent),
+            optimal_width_outer(optimal_width_outer),
+            optimal_width_inner(optimal_width_inner)
+        {
+            name = "RedistributeBeadingStrategy";
+        }
+
+        virtual ~RedistributeBeadingStrategy() override {}
+
+        Beading compute(coord_t thickness, coord_t bead_count) const override;
+
+        coord_t getOptimalThickness(coord_t bead_count) const override { return parent->getOptimalThickness(bead_count);  }
+        coord_t getTransitionThickness(coord_t lower_bead_count) const override { return parent->getTransitionThickness(lower_bead_count); }
+        coord_t getOptimalBeadCount(coord_t thickness) const override { return parent->getOptimalBeadCount(thickness); }
+        coord_t getTransitioningLength(coord_t lower_bead_count) const override { return parent->getTransitioningLength(lower_bead_count); }
+        float getTransitionAnchorPos(coord_t lower_bead_count) const override { return parent->getTransitionAnchorPos(lower_bead_count); }
+
+        virtual std::string toString() const override { return std::string("RedistributeBeadingStrategy+") + parent->toString(); }
```

Comment:
`virtual` is redundant since allready declared `override`

Reply:
I'll remove it, but I don't think override is completely redundant, as it makes it clear that this is not the base class (which also has to be virtual for these functions, but isn't 'override' there).